File: algorithms/Codility-master/Codility-master/5_PrefixSums/PassingCars.py
# 100%
def solution(A):
    # write your code in Python 3.6
    
    n = len(A)
    if n == 0:
        return 0
    
    count0 = 0
    count = 0
    for i in range(n):
        if A[i] == 0:
            count0 += 1
        if A[i] == 1:
            count += count0
        if count > 1e9:
                return -1    
            
    return count
        

File: algorithms/Codility-master/Codility-master/5_PrefixSums/PassingCars.js
// 100%
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    var n =  A.length;
    if ( n == 0){
        return -1;
    }
    
    var count = 0;
    var east = 0;
    let i;
    for ( i = 0; i<n; i++){
        if (A[i] == 0){
            east += 1;
        }
        if (A[i] == 1){
            count += east;
        }
        if (count > 1e9){
            return -1;
        }
    }
    return count;
}
